src/lib/permissions/proposals/policies/index.ts

```diff
@@ -24,6 +29,11 @@ export function proposalResolver({ resourceId }: Resource): Promise<ProposalReso
       archived: true
     }
   });
+
+  if (!proposal) {
+    throw new ProposalNotFoundError(resourceId);
```

Comment:
i really prefer we use findUniqueOrThrow(), these custom errors don't help enough for all the boilerplate. we can make the error handlers smart to detect the prisma error type

Reply:
Fair enough, this is a pattern I've been using so far, but we could indeed target these errors within our top level API handlers, and do away with the boilerplate for the DataNotFound errors